src/cc65/declare.c

```diff
@@ -457,6 +457,30 @@ static unsigned ParseOneStorageClass (void)
 
 
 
+static void CheckArrayElementType (Type* DataType)
+/* Check if data type consists of arrays of incomplete element types */
+{
+    Type* T = DataType;
+
+    while (T->C != T_END) {
+        if (IsTypeArray (T)) {
+            ++T;
+            if (IsIncompleteESUType (T)) {
+                /* We cannot have an array of incomplete elements */
+                Error ("Array of incomplete element type '%s'", GetFullTypeName (T));
+            } else if (SizeOf (T) == 0) {
+                /* We could support certain 0-size element types as an extension */
+                if (!IsTypeVoid (T) || IS_Get (&Standard) != STD_CC65) {
+                    Error ("Array of 0-size element type '%s'", GetFullTypeName (T));
+                }
+            }
+        }
+        ++T;
```

Comment:
Yep. I didn't think it could be a problem as the error would still be caught and the original error message was general enough for it. But now the error message is more specific that it would mismatch the real error. I'll change that.